Add detection of "feat. X" parts in parentheses; Fix #5436

klb2 opened the pull request:
## Description

Fixes #5436

The `ftintitle.contains_feat` function uses the regular expression `feat_tokens` from `beets/plugins.py` to match existing feat. X parts. In the old version, the regex only matched "feat. X" parts when they are preceded by a space.

This update adds matching "feat. X" parts in parentheses and brackets.

## To Do

<!--
- If you believe one of below checkpoints is not required for the change you
  are submitting, cross it out and check the box nonetheless to let us know.
  For example: - [x] ~Changelog~
- Regarding the changelog, often it makes sense to add your entry only once
  reviewing is finished. That way you might prevent conflicts from other PR's in
  that file, as well as keep the chance high your description fits with the
  latest revision of your feature/fix.
- Regarding documentation, bugfixes often don't require additions to the docs.
- Please remove the descriptive sentences in braces from the enumeration below,
  which helps to unclutter your PR description.
-->

- [ ] ~Documentation.~
- [ ] Changelog. (Add an entry to `docs/changelog.rst` to the bottom of one of the lists near the top of the document.)
- [X] Tests.

JOJ0 commented on beets/plugins.py:
```diff
@@ -518,7 +518,7 @@ def feat_tokens(for_artist=True):
     if for_artist:
         feat_words += ["with", "vs", "and", "con", "&"]
-    return r"(?<=\s)(?:{})(?=\s)".format(
+    return r"(?<=[\s,(,\[])(?:{})(?=\s)".format(
```
Not sure if my regex knowledge is enough here but it looks like with `[\s,(,\[]` you try to say "match either whitespace or bracket open or square bracket open". Why are there comma (`,`) characters in between? Are they really required?  Really not sure and did not look it up but from the top of my head this should suffice: `[\s(\[]`

klb2 replied:
Obviously yes. Thanks for spotting this. I have no idea why I put a comma in there.

klb2 commented:
> Nice one! Thanks for this fix! LGTM. Please add a short changelog entry and we merge it.

Thanks! I have fixed the unnecessary comma in the regex and added a changelog entry.
